README.md
# Carchain-Cities
A library to separate the cities in Carchain

# Installation

```bash
npm i @carchain/cities
```


# Import

#### JavaScript

```javascript
const cc = require('@carchain/cities')
//or
const {cities , countries} = require('@carchain/cities')
```

#### TypesSript

```typescript
import * as cc from '@carchain/cities';
//or
import {cities , countries} from '@carchain/cities';
```

## Usage

#### countries

```js

cc.countries.find();

/**
 * [
    {
      id: 'c1',
      code: 'IRN',
      globalName: 'Iran, Islamic Republic of',
      name: { en: 'iran', fa: 'ایران' },
      center: [ 51.337902, 35.699773 ],
      isActive: true,
      location: { type: 'Polygon', coordinates: [Array] }
    },
    ...
  ]

 */


cc.countries.findByCode('IRN');

/**
     * {
     id: 'c1',
    code: 'IRN',
    globalName: 'Iran, Islamic Republic of',
    name: { en: 'iran', fa: 'ایران' },
    center: [ 51.337902, 35.699773 ],
    isActive: true,
    location: { type: 'Polygon', coordinates: [ [Array] ] }
    }
 */


//Country code does not exist in the database
cc.countries.findByCode('IMN');

/**
 *    undefined
 */


cc.countries.findById('c1');

/**
 * {
  id: 'c1',
  code: 'IRN',
  globalName: 'Iran, Islamic Republic of',
  name: { en: 'iran', fa: 'ایران' },
  center: [ 51.337902, 35.699773 ],
  isActive: true,
  location: { type: 'Polygon', coordinates: [ [Array] ] }
}
 */


//Country ID does not exist in the database
cc.countries.findById('c2');

/**
 *     undefined
 */


cc.countries.searchPoint(1 , 1);

/**
 *  null
 */

cc.countries.searchPoint(51.427001953125, 35.67877297088322 );

/**
 * {
  id: 'c1',
  code: 'IRN',
  globalName: 'Iran, Islamic Republic of',
  name: { en: 'iran', fa: 'ایران' },
  center: [ 51.337902, 35.699773 ],
  isActive: true,
  location: { type: 'Polygon', coordinates: [ [Array] ] }
}
 */

```

#### cities


```js

cc.cities.findById('t1');
/**
{
  id: 't1',
  countryCode: 'IRN',
  countryGlobalName: 'Iran, Islamic Republic of',
  name: { en: 'tehran', fa: 'تهران' },
  center: [ 51.392904, 35.687976 ],
  location: { type: 'Polygon', coordinates: [ [Array] ] },
  isActive: true
}
 */


cc.cities.findById('t5');

/**
 * undefined
 */


cc.cities.find({isActive : true});

/**
 * 
[
  {
    id: 't1',
    countryCode: 'IRN',
    countryGlobalName: 'Iran, Islamic Republic of',
    name: { en: 'tehran', fa: 'تهران' },
    center: [ 51.392904, 35.687976 ],
    location: { type: 'Polygon', coordinates: [Array] },
    isActive: true
  },
  {
    id: 't2',
    countryCode: 'IRN',
    countryGlobalName: 'Iran, Islamic Republic of',
    name: { en: 'isfahan', fa: 'اصفهان' },
    center: [ 51.670188, 32.666113 ],
    location: { type: 'Polygon', coordinates: [Array] },
    isActive: true
  } ,
  ...
]
 */


cc.cities.find({isActive : false});

/**
 * [
  {
    id: 't3',
    countryCode: 'IRN',
    countryGlobalName: 'Iran, Islamic Republic of',
    name: { en: 'mashhad', fa: 'مشهد' },
    center: [ 59.616919, 36.291884 ],
    location: { type: 'Polygon', coordinates: [Array] },
    isActive: false
  }
]
 */


cc.cities.find({id : 't2'});
/**
 * [
  {
    id: 't2',
    countryCode: 'IRN',
    countryGlobalName: 'Iran, Islamic Republic of',
    name: { en: 'isfahan', fa: 'اصفهان' },
    center: [ 51.670188, 32.666113 ],
    location: { type: 'Polygon', coordinates: [Array] },
    isActive: true
  }
]
 */

cc.cities.searchPoint(51.427001953125 , 35.67877297088322);

/**
 * {
  id: 't1',
  countryCode: 'IRN',
  countryGlobalName: 'Iran, Islamic Republic of',
  name: { en: 'tehran', fa: 'تهران' },
  center: [ 51.392904, 35.687976 ],
  location: { type: 'Polygon', coordinates: [ [Array] ] },
  isActive: true
}
 */


cc.cities.searchPoint(1 , 1);

/**
 * null
 */


```



## Project
[کارچین](https://carchainnet.ir/)




CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [1.1.3](https://github.com/rezarazani/Carchain-Cities/compare/v1.1.2...v1.1.3) (2021-04-15)

### [1.1.2](https://github.com/rezarazani/Carchain-Cities/compare/v1.1.1...v1.1.2) (2021-04-15)


### Bug Fixes

* changing structure ([929f428](https://github.com/rezarazani/Carchain-Cities/commit/929f428170cd3983ec1e3b0f06f8a9c91fb90d8b))

### [1.1.1](https://github.com/rezarazani/Carchain-Cities/compare/v1.1.0...v1.1.1) (2021-04-15)


### Bug Fixes

* delete .npmrc ([6a5064a](https://github.com/rezarazani/Carchain-Cities/commit/6a5064a08297dcd0b821e4876ad408913f37a9ae))
* wrong country ([5e43c2a](https://github.com/rezarazani/Carchain-Cities/commit/5e43c2a9f70490c62aebeafab2c33280c10c4660))

## [1.1.0](https://github.com/rezarazani/Carchain-Cities/compare/v1.0.1...v1.1.0) (2021-04-14)


### Features

* add data ([750d448](https://github.com/rezarazani/Carchain-Cities/commit/750d44808abc66d99ee8e74ee72fa3a0dc5ce43e))

### 1.0.1 (2021-04-13)


### Bug Fixes

* commit lint ([10f1ff9](https://github.com/rezarazani/Carchain-Cities/commit/10f1ff90c7457a232fec4dcf93e61a9a91a0f143))


src/city/data/index.ts
import { City } from '../interface';

/**
 * 
 *    {"type": "Point", "coordinates": [lon, lat]
 * 
 */

export const cities: Array < City > = [
    {
        id : 't1' ,
        countryCode : 'IRN' ,
        countryGlobalName : 'Iran, Islamic Republic of',
        name : {
            en : 'tehran',
            fa : 'تهران'
        },
        center : [51.392904 ,35.687976] ,
        location :  {
          "type": "Polygon",
          "coordinates": [
              [
                  [51.258, 35.822],
                  [50.989, 35.443],
                  [51.184, 35.332],
                  [51.66, 35.728],
                  [51.578, 35.828],
                  [51.258, 35.822]
              ]
          ]
        },
        isActive : true 
    },
    {
        id : 't2' ,
        countryCode : 'IRN' ,
        countryGlobalName : 'Iran, Islamic Republic of',
        name : {
            en : 'isfahan',
            fa : 'اصفهان'
        },
        center : [51.670188 ,32.666113] ,
        location : {
          "type": "Polygon",
          "coordinates": [
              [
                  [51.406, 32.791],
                  [51.365, 32.491],
                  [51.879, 32.381],
                  [51.985, 32.801],
                  [51.406, 32.791]
              ]
          ]
      } ,
        isActive : true 
    },
    {
        id : 't3' ,
        countryCode : 'IRN' ,
        countryGlobalName : 'Iran, Islamic Republic of',
        name : {
            en : 'mashhad',
            fa : 'مشهد'
        },
        center : [59.616919 ,36.291884] ,
        location : {
          "type": "Polygon",
          "coordinates": [
              [
                  [59.344, 36.528],
                  [59.365, 36.363],
                  [59.687, 36.071],
                  [59.998, 36.299],
                  [59.722, 36.531],
                  [59.344, 36.528]
              ]
          ]
        },
        isActive : true 
    },
    

]